src/vehicle.cpp

```diff
@@ -4693,7 +4773,12 @@ veh_collision vehicle::part_collision( int part, const tripoint &p,
                 }
             }
         } else if( ret.type == veh_coll_body ) {
-            int dam = obj_dmg*dmg_mod/100;
+            int dam = obj_dmg * dmg_mod / 100;
+            if( chainsaw_on && part_info( ret.part ).has_flag( "CHAINSAW" ) &&
+                parts[part].hp > 0 ) {
+                dam += 66;
```

Comment:
I'm thinking `on_move` and `idle`, but I'm not sure how to balance it.
It shouldn't hit the same zed twice if it is invoked in both of those (for example, by moving into zed tile and ending the turn there). Maybe give the "sawed" zed an effect that would prevent further chainsaw hits during the turn? Or at least the hits occurring in `idle`.

Additionally, the saw could be damaged when it hits armored targets.